src/libostree/ostree-fetcher-curl.c

```diff
@@ -771,6 +771,9 @@ initiate_next_curl_request (FetcherRequest *req,
   if ((self->config_flags & OSTREE_FETCHER_FLAGS_TRANSFER_GZIP) > 0)
     curl_easy_setopt (req->easy, CURLOPT_ACCEPT_ENCODING, "");
 
+  /* If we have e.g. basic auth in the URL string, let's honor that */
+  curl_easy_setopt (req->easy, CURLOPT_HTTPAUTH, CURLAUTH_ANY);
```

Comment:
Take note that when the URL is set to the curl object, at some point we call `soup_uri_to_string`, which on the [documentation](https://developer.gnome.org/libsoup/stable/SoupURI.html#soup-uri-to-string) specifically says:

> Note that the output will never contain a password, even if `uri` does.

I'd suggest to set the username and password manually like this:

```suggestion
  const char *username = soup_uri_get_user (req->mirrorlist->pdata[req->idx]);
  curl_easy_setopt (req->easy, CURLOPT_USERNAME, username);

  const char *password = soup_uri_get_password (req->mirrorlist->pdata[req->idx]);
  curl_easy_setopt (req->easy, CURLOPT_PASSWORD, password);
```

With this change it works for me for a protected repo.

Reply:
This worked for me too, thanks!!

Reply:
Thanks a lot for taking the time to track that down!  I ended up using that suggestion but factored out a variable for the URI that we use once.